### PR TITLE
SALTO-2302 Use new statuses api for fetch in Jira

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Jira recently published a new [API](https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-status/#api-rest-api-3-statuses-search-get) for statuses.
In this change I change the previous statuses fetch to use the new API.
In the next PR i'll change statuses deploy as well.

---

_Release Notes_: 

Jira:
- All `Status` instances will now have all the relevant information.

---
_User Notifications_: 

Jira:
- For `Status` instances, some of the fields would change.

### CODE REVIEW

File: packages/jira-adapter/src/filters/statuses/status_deployment.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ import { deployWithJspEndpoints } from '../../deployment/jsp_deployment'
 import { STATUS_TYPE_NAME } from '../../constants'
 
 const log = logger(module)
+const statusCategoryNameToId = new Map([['TODO', 2], ['DONE', 3], ['IN_PROGRESS', 4]])
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/statuses/status_deployment.ts:25]: nit: `STATUS_CATEGORY_NAME_TO_ID`
also no need for Map, the native object will work
```suggestion
const STATUS_CATEGORY_NAME_TO_ID: Record<string, number> = {
  TODO: 2,
  ...
}
```

File: packages/jira-adapter/test/filters/statuses/status_deployment.test.ts [hunk 2]
```diff
@@ -62,32 +62,46 @@ describe('statusDeploymentFilter', () => {
   })
 
   describe('onFetch', () => {
+    // change this test
```
Comment (alonstern) [inline packages/jira-adapter/test/filters/statuses/status_deployment.test.ts:65]: nit: should this be removed?

**Review by alonstern — COMMENTED** [review]
Looks good! mostly nits
I think we also need a noise reduction for the removal of the iconUrl in the SaaS
The relevant code is in https://github.com/salto-io/salto_private/blob/master/packages/ws-operations/src/wslc/operations/fetch/env-notifications/changes-matcher.ts#L373

File: packages/jira-adapter/test/filters/statuses/status_deployment.test.ts (hunk 2 above)
Comment (shir-reifenberg) [reply packages/jira-adapter/test/filters/statuses/status_deployment.test.ts:65]: yes oops

**Review by alonstern — APPROVED** [review]
Looks good 👌 just make sure to merge after the noise reduction is merged

**Pull request merged**